ccm_web/server/src/api/api.section.handler.ts

```diff
@@ -47,10 +53,86 @@ export class SectionApiHandler {
     return enrollmentsResult.map(e => e.user.login_id)
   }
 
-  async enrollUser (user: SectionUserDto): Promise<CanvasEnrollment | APIErrorData> {
-    const enrollLoginId = user.loginId
-      .replace(/@([^@.]+\.)*umich\.edu$/gi, '')
-      .replace('@', '+')
+  async createExternalUser (user: SectionExternalUserDto, accountID: number): Promise<CanvasUser | APIErrorData> {
+    const email = user.email
+    const loginId = email.replace('@', '+')
+
+    try {
+      const endpoint = `accounts/${accountID}/users`
+      const method = HttpMethod.Post
+      const body = {
+        user: {
+          // API doesn't provide separate given- and surname fields
+          name: `${user.givenName} ${user.surname}`,
+          sortable_name: `${user.surname}, ${user.givenName}`,
+          terms_of_use: true,
+          skip_registration: true
+        },
```

Comment:
You had questioned some of your own decisions around code organization, and while generally I think the PR looks pretty solid, I was uncertain about this, so I'm providing food for thought: this action of creating external users really has nothing to do with a section (we're grouping those actions together for user convenience in the UI). I started creating these `*Handler` classes because I wanted to group together similar API interactions (usually using the same REST API URL stem, e.g. `sections/`). Would it make more sense to put this method in the `AdminApiHandler` class that already exists, or a new `AccountApiHandler`? (see https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/203). To be clear, you will still need to enroll users, and that can still be done as proposed with the modified version of the `SectionApiHandler.enrollUser` method below. Let me know if I'm not communicating my thoughts clearly on this.

Reply:
OK, I can try moving `createExternalUsers()` and `createExternalUser()` from `SectionApiHandler` to `AdminApiHandler`.

Reply:
That was a fairly simple change.  It was more work to test it still worked correctly than it did to actually make the change.

🏁

Reply:
Cool